main.ts

```diff
@@ -121,34 +130,120 @@ class TodoistPluginSettingTab extends PluginSettingTab {
 		new Setting(containerEl)
 			.setName('Enable Subtasks')
-			.setDesc("When enabled, any Subtasks associated with Todos meeting filter criteria from your keyword will be shown, indented, under the parent Todo.")
+			.setDesc("When enabled, any Subtasks associated with Tasks meeting filter criteria from your keyword will be shown, indented, under the parent Task.")
 			.addToggle(t =>
 				t.setValue(this.plugin.settings.showSubtasks)
 					.onChange(async (value) => {
 							this.plugin.settings.showSubtasks = value;
+							if (this.plugin.settings.showSubtasks == false) {
+								subtaskDiv.toggleClass('hide',true);
+							} else {
+								subtaskDiv.toggleClass('hide',false);
+							}
+							await this.plugin.saveSettings();
+						}
+					));
+		const subtaskDiv = containerEl.createEl("div");
+		subtaskDiv.addClass("todoist-setting-div");
+		new Setting(subtaskDiv)
+			.setName('Include Subtasks with no due dates')
+			.setDesc("This includes showing subtasks with no due dates that are associated with any parent tasks included in the filter criteria. On by default.")
+			.setClass("todoist-setting-top")
+			.addToggle(t =>
+				t.setValue(this.plugin.settings.noDateSubtasks)
+					.onChange(async (value) => {
+							this.plugin.settings.noDateSubtasks = value;
+							await this.plugin.saveSettings();
+						}
+					));
+		new Setting(subtaskDiv)
+			.setName('Limit Subtasks by due date assigned')
+			.setDesc("When this and Enable Subtasks is on, this filters out subtasks with due date assigned that are not today. Off by default.")
+			.setClass("todoist-setting-bottom")
+			.addToggle(t =>
+				t.setValue(this.plugin.settings.todaysSubtasks)
+					.onChange(async (value) => {
+							this.plugin.settings.todaysSubtasks = value;
+							await this.plugin.saveSettings();
+						}
+					));
+	}
+
+	private addFormatSetting(containerEl: HTMLElement) {
+		/* 
+			Task Formatting
+		*/
+		this.containerEl.addClass("todoist-setting-section");
+		containerEl.createEl('h2', {text: 'Task Formatting'}).addClass("heading");
+		const formatDescription = document.createDocumentFragment();
+		formatDescription.createEl("span", null, (span) => {
+			span.innerText = 'These settings impact how each task is formatted. Example: ';
+			span.createEl("span", null, (span) => {
+				span.innerText = "`${task.content} -- $taskPriority -- [src](${task.url}) ${description}`";
+				span.addClasses(["code"]);
+			});
+		});
+		new Setting(containerEl).setDesc(formatDescription);
+		new Setting(this.containerEl)
+			.setName('Priority')
+			.setDesc("Include Priority.")
+			.setClass("todoist-setting-top")
+			.addToggle(t =>
+				t.setValue(this.plugin.settings.showPriority)
+					.onChange(async (value) => {
+							this.plugin.settings.showPriority = value;
+							await this.plugin.saveSettings();
+						}
+					)); 
+		new Setting(this.containerEl)
+			.setName('Link')
+			.setDesc("Include Link to task in Todoist.")
```

Comment:
So the `toggleServerTaskStatus` function currently requires the link in order to work. I don't think we want to have this option silently break task toggling. So let's remove until we find a way to handle linking the source task to server task without the URL?